### PR TITLE
Problem: no integration tests for valset update

### PR DESCRIPTION
**Pull request by thomas-nguy**

### CODE REVIEW

File: integration_tests/valset_stress_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+				currentNonce, err = s.getLastValsetNonce(gravityContract)
+				if currentNonce != startingNonce {
+					return true
+				}
+			}
+			return true
+		}, 5*time.Minute, 10*time.Second, "Validator set is not yet updated")
+
+		if currentNonce != startingNonce {
+			s.T().Logf("Validator set successfully updated! nonce: %d", currentNonce)
```
Comment (tomtau) [inline integration_tests/valset_stress_test.go]: should this be asserted ? something like
`s.Truef(currentNonce != startingNonce, "Failed to update validator set")` ?

**Review by tomtau — DISMISSED** [review]

File: integration_tests/valset_stress_test.go (hunk 1 above)
Comment (thomas-nguy) [reply integration_tests/valset_stress_test.go]: yes

File: integration_tests/valset_stress_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+				return false
+			}
+			return true
+		}, 5*time.Minute, 10*time.Second, "Delegate to validator failed will retry")
+
+		// Verify that delegation went through.
+		s.T().Logf("verifying delegation")
+		s.Require().Eventuallyf(func() bool {
+
+			s.Require().NoError(err, "error querying delegator bonded validators")
```
Comment (adu-crypto) [inline integration_tests/valset_stress_test.go]: why do we assert no error before we actually query the delegation?

File: integration_tests/valset_stress_test.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+			return true
+		}, 20*time.Second, 1*time.Second, "Signerset can't be retrieved")
+
+		// Grab current nonce.
+		currentNonce, err := s.getLastValsetNonce(gravityContract)
+		s.Require().NoError(err, "error getting current nonce")
+
+		// Run a loop until you get a nonce higher than the initial nonce
+		s.Require().Eventuallyf(func() bool {
+			for currentNonce == startingNonce {
```
Comment (adu-crypto) [inline integration_tests/valset_stress_test.go]: do we need a loop to retrieve valid nonce given `Eventuallyf` itself will `periodically checking target function each tick`?

File: integration_tests/valset_stress_test.go (hunk 2 above)
Comment (thomas-nguy) [reply integration_tests/valset_stress_test.go]: Indeed this part is quite weird. I did some cleanup

**Review by tomtau — APPROVED** [review]

**Review by adu-crypto — APPROVED** [review]

**Pull request merged**